generators/csharp/sdk/src/options/RequestOptionsGenerator.ts

```diff
@@ -0,0 +1,132 @@
+import { csharp, CSharpFile, FileGenerator } from "@fern-api/csharp-codegen";
+import { join, RelativeFilePath } from "@fern-api/fs-utils";
+import { Name } from "@fern-fern/ir-sdk/api";
+import { SdkCustomConfigSchema } from "../SdkCustomConfig";
+import { SdkGeneratorContext } from "../SdkGeneratorContext";
+
+export const BASE_URL_FIELD_NAME = "BaseUrl";
+export const ENVIRONMENT_FIELD_NAME = "Environment";
+
+export const HTTP_CLIENT_FIELD = csharp.field({
+    access: "public",
+    name: "HttpClient",
+    get: true,
+    init: true,
+    type: csharp.Type.optional(
+        csharp.Type.reference(
+            csharp.classReference({
+                name: "HttpClient",
+                namespace: "System.Net.Http"
+            })
+        )
+    ),
+    summary: "The http client used to make requests."
+});
+
+export const MAX_RETRIES_FIELD = csharp.field({
+    access: "public",
+    name: "MaxRetries",
+    get: true,
+    init: true,
+    type: csharp.Type.optional(csharp.Type.integer()),
+    summary: "The http client used to make requests."
+});
+
+export const TIMEOUT_IN_SECONDS = csharp.field({
+    access: "public",
+    name: "TimeoutInSeconds",
```

Comment:
I didn't really put much thought in the `timeout` property when initially building, but probably makes more sense to align with the underlying http client? 

https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-8.0

Reply:
Great call here, especially because the option hasn't been used yet. Updated to use `System.TimeSpan`.